lib/puppet/application/device.rb

```diff
@@ -319,15 +327,12 @@ def main
             end
           else
             Puppet.info _("starting applying configuration to %{target} at %{scheme}%{url_host}%{port}%{url_path}") % { target: device.name, scheme: scheme, url_host: device_url.host, port: port, url_path: device_url.path }
-            # this will reload and recompute default settings and create the devices sub vardir
-            Puppet.settings.use :main, :agent, :ssl
-            # ask for a ssl cert if needed, but at least
-            # setup the ssl system for this device.
-            setup_host(device.name)
 
-            require 'puppet/configurer'
-            configurer = Puppet::Configurer.new
-            configurer.run(:network_device => true, :pluginsync => Puppet::Configurer.should_pluginsync? && !options[:libdir])
+            Puppet.override(ssl_context: ssl_context) do
+              require 'puppet/configurer'
```

Comment:
Why do we need this `require` here and we don't need a similar `require` before line 280?

Reply:
Good point. The device application already requires it on line 2, so it's not needed here at all. I'll remove it.